src/server/scripts/ExilesReach/zone_exiles_reach.cpp

```diff
@@ -4114,6 +4296,616 @@ class spell_tutorial_health_dnt : public SpellScript
+        {
+            Position jumpto = { 19.5174f, -2513.75f, 74.0545f, me->GetOrientation() };
+            me->RemoveAllAuras();
+            me->SetDisableGravity(false);
+            me->SetTemplateRooted(false);
+            me->GetMotionMaster()->MoveJump(jumpto, 7.9894905f, 19.29110336303710937f);
+            Talk(SAY_GET_OUT_OF_HERE);
+            _events.ScheduleEvent(EVENT_RUN_TO_PLAINS, 4s);
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+            case EVENT_RUN_TO_PLAINS:
+                me->GetMotionMaster()->MovePoint(0, 51.005207f, -2485.644f, 78.15223f);
+                me->DespawnOrUnsummon(5s);
+                break;
+            default:
+                break;
+            }
+        }
+    }
+private:
+    EventMap _events;
+};
+
+enum OgreOverseerQuilboarText
+{
+    SAY_AGGRO = 0,
+    SAY_DEATH = 1,
+};
+
+enum ExilesReachQuilboarData
+{
+    ITEM_STITCHED_CLOTH_TUNIC      = 174811,
+    ITEM_STITCHED_LEATHER_TUNIC    = 174812,
+    ITEM_LINKED_MAIL_HAUBERK       = 174813,
+    ITEM_DENTED_CHESTPLATE         = 174814,
+
+    QUEST_BRIARPATCH_CHEST_DROPPED = 58904
+};
+
+enum QuilboarWarriorGeomancer
+{
+    EVENT_CHECK_FOR_AURA       = 1,
+    EVENT_BRUTAL_STRIKE        = 2,
+    EVENT_GEOMANCER_EARTH_BOLT = 2,
+
+    SPELL_NECROTIC_BURST       = 313261,
+    SPELL_QUILBOAR_SLEEP_DNT   = 313265,
+    SPELL_BRUTAL_STRIKE        = 317383,
+    SPELL_GEOMANCER_EARTH_BOLT = 321188
+};
+
+struct npc_quilboar_warrior : public ScriptedAI
+{
+    npc_quilboar_warrior(Creature* creature) : ScriptedAI(creature) { _outOfCombat = false; }
+
+    void JustAppeared() override
+    {
+        Reset();
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+        _events.ScheduleEvent(EVENT_CHECK_FOR_AURA, 1s);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        _outOfCombat = false;
+
+        if (me->HasAura(SPELL_QUILBOAR_SLEEP_DNT))
+        {
+            me->SetReactState(REACT_AGGRESSIVE);
+            me->RemoveAura(SPELL_QUILBOAR_SLEEP_DNT);
+            me->SetStandState(UNIT_STAND_STATE_STAND);
+        }
+
+        if (!urand(0, 2))
+            Talk(SAY_AGGRO, who);
+
+        _events.ScheduleEvent(EVENT_BRUTAL_STRIKE, 3s, 5s);
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        if (!urand(0, 2))
+            Talk(SAY_DEATH, killer);
+
+        for (auto const& [playerGuid, loot] : me->m_personalLoot)
+        {
+            for (LootItem const& lootItem : loot->items)
+            {
+                if (lootItem.itemid == ITEM_STITCHED_CLOTH_TUNIC || lootItem.itemid == ITEM_STITCHED_LEATHER_TUNIC || lootItem.itemid == ITEM_LINKED_MAIL_HAUBERK || lootItem.itemid == ITEM_DENTED_CHESTPLATE)
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, playerGuid))
```

Comment:
move this check before `for (LootItem const& lootItem : loot->items)`, as we do not need to retrieve it multiple times, once is enough